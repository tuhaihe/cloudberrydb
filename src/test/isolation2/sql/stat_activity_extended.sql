-- pg_stat_activity_extended is only used in cloud with warehouse
-- In CBDB, pg_stat_activity_extended warehouse_id should always be NULL

create table test_activity(a int);

0: insert into test_activity select * from generate_series(1,100);

-- in CBDB QD/QE warehouse_id must be NULL

SELECT warehouse_id, query from pg_stat_activity_extended WHERE query LIKE 'insert into test_activity select * from generate_series(1,100);';

SELECT warehouse_id, query from gp_dist_random('pg_stat_activity_extended') WHERE query LIKE 'insert into test_activity select * from generate_series(1,100);' limit 1;

drop table test_activity;