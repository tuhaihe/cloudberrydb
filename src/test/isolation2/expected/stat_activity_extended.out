-- pg_stat_activity_extended is only used in cloud with warehouse
-- In CBDB, pg_stat_activity_extended warehouse_id should always be NULL

create table test_activity(a int);
CREATE

0: insert into test_activity select * from generate_series(1,100);
INSERT 100

-- in CBDB QD/QE warehouse_id must be NULL

SELECT warehouse_id, query from pg_stat_activity_extended WHERE query LIKE 'insert into test_activity select * from generate_series(1,100);';
 warehouse_id | query                                                           
--------------+-----------------------------------------------------------------
              | insert into test_activity select * from generate_series(1,100); 
(1 row)

SELECT warehouse_id, query from gp_dist_random('pg_stat_activity_extended') WHERE query LIKE 'insert into test_activity select * from generate_series(1,100);' limit 1;
 warehouse_id | query                                                           
--------------+-----------------------------------------------------------------
              | insert into test_activity select * from generate_series(1,100); 
(1 row)

drop table test_activity;
DROP
